include/aspect/material_model/interface.h

```diff
@@ -778,8 +778,8 @@ namespace aspect
          * inputs in @p in. If MaterialModelInputs.strain_rate has the length
          * 0, then the viscosity does not need to be computed.
          */
-        virtual void evaluate(const MaterialModelInputs &in,
-                              MaterialModelOutputs &out) const = 0;
+        virtual void evaluate(const typename MaterialModel::MaterialModelInputs<dim> &in,
```

Comment:
The typename here is wrong. There are likely many locations for this -- can you change them with a regex replace?